`fallthrough` may be redefined (as in OpenZFS's spl), so use `__fallthrough`

--- boost_1_79_0/boost/config/compiler/gcc.hpp.orig	2022-06-04 19:46:23.536373946 +0200
+++ boost_1_79_0/boost/config/compiler/gcc.hpp	2022-06-08 17:18:20.938885867 +0200
@@ -320,7 +320,7 @@
 #endif
 
 #if __GNUC__ >= 7
-#  define BOOST_FALLTHROUGH __attribute__((fallthrough))
+#  define BOOST_FALLTHROUGH __attribute__((__fallthrough__))
 #endif
 
 #if (__GNUC__ < 11) && defined(__MINGW32__) && !defined(__MINGW64__)
